linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/Printers.scala

```diff
@@ -741,6 +751,10 @@ object Printers {
         print("]")
     }
 
+    protected def print(printedTree: PrintedTree): Unit =
+      // TODO: Indent this?
```

Comment:
This produces ugly (but correct results).

Going forward, I'd like to experiment with using the written files themselves as caches for the printed trees. Basically, once the `PrintedTree` is written, just keep its position in the output file. If the content is required, get it from the file (ideally in a streaming fashion of course).

This would make indentation even harder to deal with.

That being said, I'm not sure we're willing to sacrifice proper indentation for that :-/

WDYT?

Reply:
How ugly are we talking about? Is it that `class` members are indented at 0 instead of 2, but the bodies of methods are still correctly indented? Or are even the bodies of methods inconsistently indented?

If the latter, I don't think it's acceptable. If only for ourselves when we have to read the produced .js during development/debugging.

If the former, it seems fine. But also in that case we can probably figure out in advance that we should print them with 2-space indentation, so we could configure the printer to start at 2?

Reply:
Yes, it's just the indentation for classes and yes, we can likely figure out the required nesting in advance.